scripts/koji/koji_build.py

```diff
@@ -36,10 +46,17 @@ def check_git_repo(dirpath):
     with cd(dirpath):
         return subprocess.run(['git', 'diff-index', '--quiet', 'HEAD', '--']).returncode == 0
 
-def check_commit_is_available_remotely(dirpath, hash):
+def check_commit_is_available_remotely(dirpath, hash, target):
     with cd(dirpath):
         if not subprocess.check_output(['git', 'branch', '-r', '--contains', hash]):
             raise Exception("The current commit is not available in the remote repository")
+        expected_branch = PROTECTED_TARGETS.get(target)
+        if (
+            expected_branch is not None
+            and not is_remote_branch_commit(dirpath, hash, expected_branch)
+        ):
+            raise Exception(f"The current commit is not the last commit in the remote branch {expected_branch}.\n"
```

Comment:
I have doubts about the heuristics. We could try with this one, but I think there must be an option to force-build. Especially with `fasttrack`, where there are cases when we may have to build from a branch because master has too many changes and we want to push an update out with just a tiny XSA fix.